core/station/impl/src/migration.rs

```diff
@@ -135,16 +149,503 @@ fn apply_migration() {
+        })
+    }
+}
+
+impl<'de> Deserialize<'de> for Account {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[allow(dead_code)]
+        #[derive(Debug, Deserialize)]
+        struct PreMigrationAccount {
+            pub id: AccountId,
+            pub name: String,
+            pub metadata: Metadata,
+            pub transfer_request_policy_id: Option<UUID>,
+            pub configs_request_policy_id: Option<UUID>,
+            pub last_modification_timestamp: Timestamp,
+
+            // removed fields
+            pub balance: Option<Option<AccountBalance>>,
+            pub blockchain: Option<Blockchain>,
+            pub address: Option<String>,
+            pub standard: Option<BlockchainStandard>,
+            pub symbol: Option<String>,
+            pub decimals: Option<u32>,
+
+            // new fields
+            pub seed: Option<AccountSeed>,
+            pub assets: Option<Vec<AccountAsset>>,
+            pub addresses: Option<Vec<AccountAddress>>,
+        }
+
+        let pre_migration_entry = PreMigrationAccount::deserialize(deserializer)?;
+
+        #[cfg(test)]
+        if pre_migration_entry.seed.is_none() {
+            MIGRATED_ENTRIES.with(|entries| {
+                *entries.borrow_mut() += 1;
+            });
+
+            if let Some(address) = &pre_migration_entry.address {
+                MIGRATED_ACCOUNTS.with(|accounts| {
+                    accounts
+                        .borrow_mut()
+                        .push((pre_migration_entry.id, address.clone()));
+                });
+            }
+        }
+
+        let seed = pre_migration_entry.seed.unwrap_or(pre_migration_entry.id);
+
+        Ok(Account {
+            id: pre_migration_entry.id,
+            name: pre_migration_entry.name,
+            metadata: pre_migration_entry.metadata,
+            transfer_request_policy_id: pre_migration_entry.transfer_request_policy_id,
+            configs_request_policy_id: pre_migration_entry.configs_request_policy_id,
+            last_modification_timestamp: pre_migration_entry.last_modification_timestamp,
+            seed,
+            assets: pre_migration_entry.assets.unwrap_or(vec![AccountAsset {
+                asset_id: INITIAL_ICP_ASSET_ID,
+                balance: pre_migration_entry.balance.unwrap_or(None),
+            }]),
+            addresses: pre_migration_entry.addresses.unwrap_or_else(|| {
+                vec![
+                    AccountAddress {
+                        address: pre_migration_entry.address.unwrap_or(
+                            InternetComputer::create().generate_account_identifier(&seed),
+                        ),
+                        format: AddressFormat::ICPAccountIdentifier,
+                    },
+                    AccountAddress {
+                        address: InternetComputer::create().generate_icrc1_address(&seed),
```

Comment:
nit: maybe instantiate `InternetComputer::create()` once before the vec and then reuse in both account addresses